python/lsst/daf/butler/_dataset_type.py

```diff
@@ -59,7 +59,7 @@ class SerializedDatasetType(BaseModel):
 
     name: StrictStr
     storageClass: StrictStr | None = None
-    dimensions: SerializedDimensionGraph | list[StrictStr] | None = None
+    dimensions: list[StrictStr] | None = None
```

Comment:
That's right, but we've been writing this with `list` for a while now, so butler server should be fine, and I figured any graphs old enough to suffer from this would probably be impossible to read already due to other incompatibilities (though I don't think I can prove that in general).